plots/dashboard/dashboard.js

```diff
@@ -0,0 +1,254 @@
+    chart.appendChild(button);
+
+    const container = document.getElementById('charts');
+    container.appendChild(chart);
+    return chart.id;
+  }
+
+  _onFocusChart(data, title) {
+    const overlay = document.createElement('div');
+    overlay.id = 'overlay';
+    document.body.appendChild(overlay);
+
+    document.getElementById('charts').style.display = 'none';
+
+    const closeButton = document.createElement('button');
+    closeButton.className = 'dth-button close-button';
+    closeButton.appendChild(document.createTextNode('Close'));
+    closeButton.addEventListener('click', onCloseFocusedChart, false);
+    overlay.appendChild(closeButton);
+
+    const chart = document.createElement('div');
+    chart.className = 'chart';
+    overlay.appendChild(chart);
+    this._renderFocusedChart(data, title, chart);
+
+    function onCloseFocusedChart() {
+      document.getElementById('charts').style.display = 'block';
+      document.body.removeChild(overlay);
+    }
+  }
+
+  _renderFocusedChart(data, title, element) {
+    Plotly.newPlot(
+      element,
+      data,
+      Object.assign({title}, this._layout, {
+        width: document.body.clientWidth - 100,
+        height: 500
+      })
+    );
+  }
+}
+
+class RenderingScheduler {
+  constructor() {
+    this._queue = [];
+  }
+
+  enqueue(fn) {
+    const isFirst = this._queue.length == 0;
+    this._queue.push(fn);
+    if (isFirst) {
+      this._render();
+    }
+  }
+
+  _render() {
+    window.requestAnimationFrame(_ => {
+      const plotFn = this._queue.shift();
+      if (plotFn) {
+        plotFn();
+        this._render();
+      }
+    });
+  }
+}
+
+const Utils = {
+  /**
+ * @param {!Element} parent
+ */
+  removeChildren(parent) {
+    while (parent.firstChild) {
+      parent.removeChild(parent.firstChild);
+    }
+  },
+
+  /**
+ * Calculate the value at a given percentile
+ * Based on: https://gist.github.com/IceCreamYou/6ffa1b18c4c8f6aeaad2
+ * @param {!Array<number>} array
+ * @param {number} percentile should be from 0 to 1
+ */
+  calculatePercentile(array, percentile) {
+    if (array.length === 0) {
+      return 0;
+    }
+    if (percentile <= 0) {
+      return array[0];
+    }
+    if (percentile >= 1) {
+      return array[array.length - 1];
+    }
+    const sorted = array.slice().sort((a, b) => a - b);
+
+    const index = (sorted.length - 1) * percentile;
+    const lower = Math.floor(index);
+    const upper = lower + 1;
+    const weight = index % 1;
+
+    if (upper >= sorted.length) {
+      return sorted[lower];
+    }
+    return sorted[lower] * (1 - weight) + sorted[upper] * weight;
+  }
+};
+
+function main() {
+  /**
+   * Navigation Start is usually not a very informative metric.
+   */
+  const metrics = Object.keys(dashboardResults).filter(m => m !== 'Navigation Start');
+
+  const renderingScheduler = new RenderingScheduler();
+  const charts = new Charts(renderingScheduler);
+  const dashboard = new Dashboard(metrics, charts);
```

Comment:
niiiiice muuuuch easier to follow!

minor nit, personally I like to separate the rendering logic from a constructor to an explicit `render()` method so it's clear what attaches to the DOM and what does basic setup and whatnot (if there's no need to keep instance variables at all some could even be static and the classes just used for logical grouping but totally up to you)